src/main/controller/UserController.ts

```diff
@@ -162,6 +162,20 @@ export class UserController {
         const userRepository = transactionalEntityManager.getCustomRepository(
           UserRepository
         );
+
+        const authenticatedUser = await userRepository.findByFirebaseId(
+          res.locals.firebaseUserId,
+          {
+            cache: true,
+          }
+        );
+
+        // todo: handle null authenticatedUser case
+
+        if (authenticatedUser?.id !== id) {
+          // todo: throw error
+        }
```

Comment:
Yup, exactly the idea we were having in mind, great job! We don't need the cache since the user _should_ be deleted.